pkg/recipes/driver/terraform.go

```diff
@@ -135,13 +135,28 @@ func (d *terraformDriver) Delete(ctx context.Context, opts DeleteOptions) error
 			logger.Info(fmt.Sprintf("Failed to cleanup Terraform execution directory %q. Err: %s", requestDirPath, err.Error()))
 		}
 	}()
-
+	// Add credential information to .gitconfig if module source is of type git.
+	if strings.HasPrefix(opts.Definition.TemplatePath, "git::") && !reflect.DeepEqual(opts.BaseOptions.Secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
+		err := addSecretsToGitConfig(opts.BaseOptions.Secrets, &opts.Recipe, opts.Definition.TemplatePath)
+		if err != nil {
+			return err
```

Comment:
We might have discussed this already.. Could you remind me if there's an issue with mocking a dependency to add a unit test here? If not, and we're mainly skipping them since it's covered in gitconfig tests, then let's add them at least for the error scenario, so that a future change doesn't accidentally ignore the error.